libs/common/src/lib/deep-freeze.spec.ts

```diff
@@ -0,0 +1,87 @@
+import { deepFreeze } from './deep-freeze';
+
+// Created a cool set of unit tests with ChatGPT: https://chat.openai.com/share/f2a078c9-e996-4d4e-8c4c-725ca2e9cafd
+
+describe('deepFreeze', () => {
+    it('should make object immutable', () => {
+        const obj = { count: 1 };
+        deepFreeze(obj);
+        expect(Object.isFrozen(obj)).toBe(true);
+        expect(() => (obj.count = 2)).toThrow();
+
+        const obj2 = {
+            nestedCounter: {
+                count: 1,
+            },
+        };
+        deepFreeze(obj2);
+        expect(Object.isFrozen(obj2.nestedCounter)).toBe(true);
+        expect(() => (obj2.nestedCounter.count = 2)).toThrow();
+    });
+
+    it('should freeze a function object', () => {
+        const fn = (): number => {
+            return 42;
+        };
+        expect(() => deepFreeze(fn)).not.toThrow();
+        // Test that the function object is frozen
```

Comment:
This comment seems obsolete?

Reply:
I removed two not useful comments. The others have at least a little bit extra information which could be useful.